scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+                filename=LOG_FILE,
+                filemode='a',
+                format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                datefmt='%H:%M:%S',
+                level=logging.DEBUG)
+
+        self._logger = logging.getLogger('supervisor')
+
+        self._logger.info(f"Initializing supervisor (pid: {os.getpid()})")
+
+        self._app_name = app_name
+        self._config = {}
+        self._nodes = {}
+        self._error = None
+
+        with open(PID_FILE, "w") as f:
+            f.write(str(os.getpid()))
+
+        self._logger.info("Opening command FIFOs")
+
+        self._in = open(FIFO_IN, "r")
+        self._out = open(FIFO_OUT, "w")
+
+        self._load_config()
+
+        self._current_index = max([0] + list(self._config["nodes"].keys()))
+
+        self._logger.info("Initialization complete")
+
+    def _stop_nodes(self):
+        self._logger.info(f"Stopping nodes")
+        for node_index in list(self._nodes.keys()):
+            self.stop_node(node_index=node_index)
+
+    def __del__(self):
+        self._stop_nodes()
+
+        self._logger.info(f"Cleaning up")
+        self._in.close()
+        self._out.close()
+        os.unlink(PID_FILE)
+        os.unlink(FIFO_IN)
+        os.unlink(FIFO_OUT)
+
+        self._logger.info(f"Supervisor shut down")
+
+    def _save_config(self):
+        with open(CONFIG_FILE, "w") as f:
+            json.dump(self._config, f, indent=4)
+
+    def _get_next_index(self):
+        self._current_index += 1
+        return self._current_index
+
+    def _make_success(self, content = ""):
+        return {"success": True, "content": str(content)}
+
+    def _make_fail(self, err):
+        return {"success": False, "content": str(err)}
+
+    def _load_config(self):
+        self._logger.info(f"Loading config from {CONFIG_FILE}")
+        try:
+            with open(CONFIG_FILE, 'r') as f:
+                self._config = json.load(f)
+            # JSON mandates str map keys, convert to int for convenience and consistency
+            self._config["nodes"] = {int(k): v for k, v in self._config["nodes"].items()}
+        except Exception as e:
+            self._error = f"Failed to load configuration from {CONFIG_FILE}: {str(e)}"
+            self._logger.error(self._error)
+            return self._make_error(self._error)
+        self._logger.debug(f"Loaded config: {json.dumps(self._config, indent=4)}")
+        return self._make_success()
+
+    def _get_node_config_item(self, i, key):
+        node_config = self._config["nodes"][i]
+        if key in node_config:
+            return node_config[key]
+
+        return self._config["default_scylla_config"][key]
+
+    def _send_reply(self, res = None):
+        if res is None:
+            res = self._make_success()
+        self._out.write(json.dumps(res) + '\n')
+        self._out.flush()
+
+    def start_cluster(self, **kwargs):
+        self._logger.info(f"start_cluster({kwargs})")
+        pass
```

Comment:
i don't quite understand this one. is this supposed to be a stub?

Reply:
Frogot to remove, not needed anymore.

Reply:
Done in v2.